src/Bicep.Core/Emit/InlineDependencyVisitor.cs

```diff
@@ -198,29 +198,44 @@ private void VisitPropertyAccessSyntaxInternal(PropertyAccessSyntax syntax)
                 return;
             }
 
-            static bool ShouldSkipInlining(ObjectType objectType, string propertyName)
+            static bool ShouldSkipInlining(ObjectType objectType, string propertyName, ResourceDeclarationSyntax? resourceSyntax = null)
```

Comment:
Since you've already got the resource symbol, should be simpler to just use that and the `resourceSymbol.SafeGetBodyProperty(...)` function instead.

Reply:
Makes sense. Made the suggested change.